vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SearchManager.java

```diff
@@ -0,0 +1,270 @@
+        Page<ISlide> slidePage = searchSlides(requestedPageForSlide, searchTerm);
+        int totalSlidePage = slidePage.getTotalPages();
+        /*
+         * spring will just return an empty dataset if a page that is greater than the
+         * total number of pages is returned. So, the page given (except when it's < 1,
+         * then set it to 1) and then checking if the total page count is smaller than
+         * the given one (if so, make another request). In most cases the page number
+         * will be within the range of pages, so we would need only one db call. Only in
+         * cases where the page number is wrong, would a second one be needed.if
+         * page>total pages,last page is returned.
+         */
+        if (page > totalSlidePage) {
+            totalSlidePage = totalSlidePage == 0 ? 1 : totalSlidePage;
+            requestedPageForSlide = PageRequest.of(totalSlidePage - 1, pageSize);
+            slidePage = searchSlides(requestedPageForSlide, searchTerm);
+        }
+        return slidePage;
+    }
+
+    /**
+     * Method to return the requested slides whose text blocks contains the search
+     * string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return list of slides whose text blocks contains the search string in the
+     *         requested page.
+     */
+    @Override
+    public Page<ISlide> searchSlideTextsAndPaginate(String searchTerm, int page) {
+        /* if page<1, 1st page is returned */
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForSlideText = PageRequest.of(page - 1, pageSize);
+        Page<ISlide> slidetextPage = searchSlideTexts(requestedPageForSlideText,
+                searchTerm);
+        
+        int totalSlideTextPage = slidetextPage.getTotalPages();
+        /*
+         * spring will just return an empty dataset if a page that is greater than the
+         * total number of pages is returned. So, the page given (except when it's < 1,
+         * then set it to 1) and then checking if the total page count is smaller than
+         * the given one (if so, make another request). In most cases the page number
+         * will be within the range of pages, so we would need only one db call. Only in
+         * cases where the page number is wrong, would a second one be needed.if
+         * page>total pages,last page is returned.
+         */
+        if (page > totalSlideTextPage) {
+            totalSlideTextPage = totalSlideTextPage == 0 ? 1 : totalSlideTextPage;
+            requestedPageForSlideText = PageRequest.of(totalSlideTextPage - 1, pageSize);
+            slidetextPage = searchSlideTexts(requestedPageForSlideText,
+                    searchTerm);
+        }
+        return slidetextPage;
+    }
+    
+
+
+    
+    @Override
+    public SearchModuleResults getSearchModuleResults(List<IModule> moduleList) {
```

Comment:
what is this method for? Why does it require an extra call? Whoever uses `SearchManager` should only have to make one call to get results.